lighthouse-core/driver/gatherers/media-sized.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gather = require('./gather');
+
+/* global document, URL, __returnResults */
+
+/* istanbul ignore next */
+function getUnsizedMediaElements() {
+  var hasSizeAttributes = function(element) {
+    // Simple early exit for an element with declared attributes.
+    if (element.getAttribute('width') && element.getAttribute('height')) {
+      return true;
+    }
+
+    // For all other cases you need to step through CSS rules and find the ones
+    // that match the element in question, then interrogate it for width & height
+    // properties
+    var rules = [];
+    for (let sheet of document.styleSheets) {
+      for (let rule of sheet.rules) {
+        if (element.matches(rule.selectorText)) {
+          rules.push(rule);
+        }
+      }
+    }
+
+    var definesSizes = rules.find(r => r.style.width !== '' && r.style.height !== '');
+    return (definesSizes !== undefined);
+  };
+
+  var unsizedElements =
+      [...document.querySelectorAll('img, video, svg')]
+      .filter(el => !hasSizeAttributes(el))
```

Comment:
A number of our gatherers report a bit more verbosely, and let the audits do the conditionals. I'd be into reporting an object for each img/video/svg element with width/height properties. And then leave it to the audit to evaluate if the sizes are defined.

That would save us some work when doing an audit for checking media elements for autoplay, or something.

But it's not a big deal for me, and we can refactor this bit later as well.